app/js/components/orderEntry/OrderEntryPage.jsx

```diff
@@ -48,11 +57,92 @@ export class OrderEntryPage extends PureComponent {
     </p>
   );
 
-  switchOrderType = (newOrderType) => {
-    if (!newOrderType) {
-      return this.props.setSelectedOrder({ currentOrderType: {} });
-    }
-    return this.props.setSelectedOrder({ currentOrderType: newOrderType });
+  handleDrugOrderData = (order) => {
+    const {
+      drugDispensingUnits,
+      drugDosingUnits,
+      drugRoutes,
+      durationUnits,
+      orderFrequencies,
+    } = this.props.configurations;
+    const {
+      action,
+      reason,
+      careSetting,
+      dose,
+      dosingUnit,
+      drugInstructions,
+      dosingType,
+      drug,
+      duration,
+      durationUnit,
+      frequency,
+      orderer,
+      dispensingQuantity,
+      dispensingUnit,
+      route,
+      type,
+      previousOrder,
+    } = order;
+
+    return {
+      action,
+      asNeeded: (reason && true) || false,
+      asNeededCondition: reason,
+      autoExpireDate: null,
+      orderReasonNonCoded: null,
+      careSetting,
+      commentToFulfiller: '',
+      dose: dose || null,
+      doseUnits: (dosingUnit && this.getUUID(drugDosingUnits, dosingUnit).uuid) || null,
+      dosingInstructions: drugInstructions || null,
+      dosingType,
+      drug,
+      duration: duration || null,
+      durationUnits: (durationUnit && this.getUUID(durationUnits, durationUnit).uuid) || null,
+      frequency: (frequency && this.getUUID(orderFrequencies, frequency).uuid) || null,
+      numRefills: 0,
+      orderer,
+      previousOrder,
+      quantity: dispensingQuantity || null,
+      quantityUnits:
+        (dispensingUnit && this.getUUID(drugDispensingUnits, dispensingUnit).uuid) || null,
+      route: (route && this.getUUID(drugRoutes, route).uuid) || null,
+      type,
+    };
+  };
+
+  handleLabOrderData = order => ({
```

Comment:
Just like https://github.com/openmrs/openmrs-owa-orderentry/pull/180/files#r216264328.
`handleLabOrderData` can be better named `formatLabOrderData` since that is what it is really doing.

Reply:
addressed